src/Amazon.DynamoDb/Actions/DynamoQuery.cs

```diff
@@ -79,18 +82,23 @@ public DynamoQuery Filter(params Expression[] conditions)
         {
             if (_filter is null)
             {
-                ExpressionAttributeNames ??= new JsonObject();
+                Dictionary<string, string> attrNames = ExpressionAttributeNames ?? new Dictionary<string, string>();
                 ExpressionAttributeValues ??= new AttributeCollection();
 
-                _filter = new DynamoExpression(ExpressionAttributeNames, ExpressionAttributeValues);
+                _filter = new DynamoExpression(attrNames, ExpressionAttributeValues);
             }
 
             foreach (Expression condition in conditions)
             {
                 _filter.Add(condition);
+
             }
 
             this.FilterExpression = _filter.Text;
+            if (_filter.HasAttributeNames && ExpressionAttributeNames == null)
+            {
+                ExpressionAttributeNames = _filter.AttributeNames;
+            }

```

Comment:
To avoid this weirdness we could add a custom `Dictionary<string, string>` converter that does not serialize Dictionaries if they are empty. As it is, I have to make sure that if its empty then the field is null. I guess maybe we could do something like have a backing field that is the dictionary with `[JsonIgnore]`, and then a property `ExpressionAttributeNames` that returns null if it's empty, like:

```
[JsonIgnore]
public Dictionary<string, string> ExpressionNamesNotNull { get; set; } = new Dictionary<string, string>();

public Dictionary<string, string>? ExpressionAttributeNames => ExpressionNamesNotNull.Count > 0 ? ExpressionNamesNoteNull : null;